src/Security/Authentication/Core/src/AuthenticationMiddleware.cs

```diff
@@ -48,17 +48,22 @@ public async Task Invoke(HttpContext context)
                 }
             }
 
+            AuthenticateResult result = null;
             var defaultAuthenticate = await Schemes.GetDefaultAuthenticateSchemeAsync();
             if (defaultAuthenticate != null)
             {
-                var result = await context.AuthenticateAsync(defaultAuthenticate.Name);
+                result = await context.AuthenticateAsync(defaultAuthenticate.Name);
                 if (result?.Principal != null)
                 {
                     context.User = result.Principal;
                 }
             }
 
+            context.Features.Set<IAuthenticationResultFeature>(new AuthenticationResultFeature
```

Comment:
As an optimization, we should implement this feature in the servers as well. It's usually a file that needs to be updated.

Reply:
That mainly works if you're setting a property on an existing feature, not assigning a new feature.

Reply:
Why can't this check for the feature?